src/vs/workbench/contrib/terminal/browser/links/terminalWordLinkProvider.ts

```diff
@@ -145,36 +152,64 @@ export class TerminalWordLinkProvider extends TerminalBaseLinkProvider {
 		// If any of the names of the folders in the workspace matches
 		// a prefix of the link, remove that prefix and continue
 		this._workspaceContextService.getWorkspace().folders.forEach((folder) => {
-			if (link.substr(0, folder.name.length + 1) === folder.name + (isWindows ? '\\' : '/')) {
+			if (link.substr(0, folder.name.length + 1) === folder.name + pathSeparator) {
 				link = link.substring(folder.name.length + 1);
 				return;
 			}
 		});
 
-		const sanitizedLink = link.replace(/:\d+(:\d+)?$/, '');
-		let exactMatch = await this._getExactMatch(sanitizedLink, link);
-		if (exactMatch) {
-			// If there was exactly one match, open it
-			const match = link.match(/:(\d+)?(:(\d+))?$/);
-			const startLineNumber = match?.[1];
-			const startColumn = match?.[3];
-			await this._editorService.openEditor({
-				resource: exactMatch,
-				options: {
-					pinned: true,
-					revealIfOpened: true,
-					selection: startLineNumber ? {
-						startLineNumber: parseInt(startLineNumber),
-						startColumn: startColumn ? parseInt(startColumn) : 0
-					} : undefined
-				}
-			});
-			return;
+		let matchLink = link;
+		if (this._capabilities.has(TerminalCapability.CwdDetection)) {
+			matchLink = this._updateLinkWithRelativeCwd(y, link, pathSeparator);
+		}
+		const sanitizedLink = matchLink.replace(/:\d+(:\d+)?$/, '');
+		try {
+			const exactMatch = await this._getExactMatch(sanitizedLink, matchLink);
+			if (exactMatch) {
+				// If there was exactly one match, open it
+				const match = matchLink.match(/:(\d+)?(:(\d+))?$/);
+				const startLineNumber = match?.[1];
+				const startColumn = match?.[3];
+				await this._editorService.openEditor({
+					resource: exactMatch,
+					options: {
+						pinned: true,
+						revealIfOpened: true,
+						selection: startLineNumber ? {
+							startLineNumber: parseInt(startLineNumber),
+							startColumn: startColumn ? parseInt(startColumn) : 0
+						} : undefined
+					}
+				});
+				return;
+			}
+		} catch {
+			// Fallback to searching quick access
+			return this._quickInputService.quickAccess.show(link);
 		}
-		// Fallback to searching quick access
 		return this._quickInputService.quickAccess.show(link);
 	}
 
+	private _updateLinkWithRelativeCwd(y: number, link: string, pathSeparator: string): string {
```

Comment:
Add some jsdoc here to explain that it uses available command information to change cwd depending on the line.